### ISSUE TITLE
Exception in yaml_to_mux plugin initialize

### ISSUE THREAD

**Issue by rbeims** [avocado-framework/avocado#2790]
I'm just starting using avocado, and I stumbled across this initialization issue:
```
Error running method "initialize" of plugin "yaml_to_mux": generator raised StopIteration
Error running method "update_defaults" of plugin "yaml_to_mux": generator raised StopIteration
Error running method "__len__" of plugin "yaml_to_mux": 'YamlToMux' object has no attribute 'variant_ids'

Variant :    /
```

The command I'm running is this:
```avocado variants -m remote_connection_github.yaml -c```

The yaml file being used is this:
```
dut_options:
    default:
        remote_host: "ipaddress"
        remote_port: "443"
        user_password: "userpw"
        admin_password: "adminpw"
```

I have tried to run tests using this file, but I always get the result reported in the beginning. 
Am I missing something?

**Comment by rbeims** [comment]
I don't know if it's relevant for this case, but I'm using avocado with python3.

**Comment by clebergnu** [comment]
Hi @rbeims , we do have a couple of issues with YAML to Mux on Python 3, but they *should* only happen if the input given is not ascii.  These issues are tracked here:

https://trello.com/c/Q8QVmj8E/1254-bug-non-ascii-character-breaks-yaml2mux

I've tried your example in the latest master:

```
$ git show 
commit cca6b17329d18b9b9b8ee667c1a5d1c46bf6871e (HEAD -> master, origin/master, origin/HEAD)
Merge: 888a7ab3 5a9a6c06
Author: Cleber Rosa <crosa@redhat.com>
Date:   Tue Aug 21 20:50:48 2018 -0400

    Merge remote-tracking branch 'cacarrara/enable-more-pylint-checks'
    
    Signed-off-by: Cleber Rosa <crosa@redhat.com>
```

And it seems to work:

```
$ avocado variants -m /tmp/t.yml -c
Multiplex variants (1):

Variant default-3df9:    /run/dut_options/default
    /run/dut_options/default:admin_password => adminpw
    /run/dut_options/default:remote_host    => ipaddress
    /run/dut_options/default:remote_port    => 443
    /run/dut_options/default:user_password  => userpw

$ avocado run -m /tmp/t.yml -- examples/tests/passtest.py
JOB ID     : 2454518133937c41261cf6a4bdb3bcfb5f53852e
JOB LOG    : /home/cleber/avocado/job-results/job-2018-08-22T10.45-2454518/job.log
 (1/1) examples/tests/passtest.py:PassTest.test;default-3df9: PASS (0.01 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.15 s
JOB HTML   : /home/cleber/avocado/job-results/job-2018-08-22T10.45-2454518/results.html
```

My first guess is that you're running the version 63.0 or earlier, on Python 3.7.  There's a fix on the latest master ( 023e3186f6577d3292941c4634ca04effef1e713 ), which will make into 64.0 which fixes the issue I believe you're having.
